.github/workflows/linting.yml

```diff
@@ -0,0 +1,54 @@
+name: Linting
+
+on: [push, pull_request]
+
+jobs:
+  pylint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+    - name: Install pylint
+      run: |
+        python -m pip install --upgrade pip
+        pip install pylint
+    - name: Run pylint
+      run: |
+        pylint algorithmic_efficiency
+        pylint sample_submissions
+        pylint tests
+
+  isort:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+    - name: Install isort
+      run: |
+        python -m pip install --upgrade pip
+        pip install isort
+    - name: Run isort
+      run: |
+        isort . --check
+
+  yapf:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+    - name: Install yapf
+      run: |
+        python -m pip install --upgrade pip
+        pip install yapf
+    - name: Run yapf
+      run: |
+        yapf . --diff --recursive
```

Comment:
can we add `--style=yapf` to this? I think that is the style that is most similar to our internal formatter

Reply:
The `yapf` configuration sits in the `setup.cfg` file (https://github.com/fsschneider/algorithmic-efficiency/blob/ada14168b738c99f640508f6a7959945c3e939c4/setup.cfg#L82-L86). As far as I know, yapf automatically uses this configuration so adding `--style=yapf` to it here is unnecessary. Using the configuration in `setup.cfg` has the advantage of one central place to define our coding style. Running it locally, either via the command line or in an editor should by default also use this configuration.

Originally, I was using the `google` style, with `column_limit=80` and `indent_width=2`. Changed it now to `yapf` style without any further modifications.